src/main/scala/viper/gobra/ast/internal/Program.scala

```diff
@@ -172,6 +187,10 @@ case class SafeTypeAssertion(resTarget: LocalVar, successTarget: LocalVar, expr:
 
 case class FunctionCall(targets: Vector[LocalVar], func: FunctionProxy, args: Vector[Expr])(val info: Source.Parser.Info) extends Stmt
 case class MethodCall(targets: Vector[LocalVar], recv: Expr, meth: MethodProxy, args: Vector[Expr])(val info: Source.Parser.Info) extends Stmt
+// Go function and method calls must receive a Member (more precisely, a fully desugared function or method)
+// in order to retrieve its pre-condition
+case class GoFunctionCall(func: FunctionMember, args: Vector[Expr])(val info: Source.Parser.Info) extends Stmt
+case class GoMethodCall(recv: Expr, meth: MethodMember, args: Vector[Expr])(val info: Source.Parser.Info) extends Stmt
```

Comment:
This might be bad, but I am not entirely certain how case classes behave. 
I would be worried about the case where a function calls itself with a go statement. 
Do when then have an infinite structure? At least, we would not have a tree anymore.

Reply:
yes, this causes an infinite loop indeed! I added a test for this case and I solved the issue by adding a flag to the function and method desugarer to ignore the body when desugaring.